ui/apps/platform/src/Containers/NetworkGraph/components/stylesComponentFactory.tsx

```diff
@@ -0,0 +1,168 @@
+
+interface EdgeProps {
+    element: Edge;
+}
+
+const contextMenuItem = (label: string, i: number): React.ReactElement => {
+    if (label === '-') {
+        return <ContextMenuSeparator key={`separator:${i.toString()}`} />;
+    }
+    return (
+        // eslint-disable-next-line no-alert
+        <ContextMenuItem key={label} onClick={() => alert(`Selected: ${label}`)}>
+            {label}
+        </ContextMenuItem>
+    );
+};
+
+const createContextMenuItems = (...labels: string[]): React.ReactElement[] =>
+    labels.map(contextMenuItem);
+
+const defaultMenu = createContextMenuItems('First', 'Second', 'Third', '-', 'Fourth');
+
+const stylesComponentFactory: ComponentFactory = (kind: ModelKind, type: string): any => {
+    if (kind === ModelKind.graph) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        return withDndDrop(graphDropTargetSpec([NODE_DRAG_TYPE]))(withPanZoom()(GraphComponent));
+    }
+    switch (type) {
+        case 'node':
+            return withCreateConnector((source: Node, target: Node | Graph): void => {
+                let targetId;
+                const model = source.getController().toModel();
+                if (isNode(target)) {
+                    targetId = target.getId();
+                } else {
+                    return;
+                }
+                const id = `e${source.getGraph().getEdges().length + 1}`;
+                if (!model.edges) {
+                    model.edges = [];
+                }
+                model.edges.push({
+                    id,
+                    type: 'edge',
+                    source: source.getId(),
+                    target: targetId,
+                });
+                source.getController().fromModel(model);
+            })(
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-ignore
+
+                withDndDrop(
+                    nodeDropTargetSpec([
+                        CONNECTOR_SOURCE_DROP,
+                        CONNECTOR_TARGET_DROP,
+                        CREATE_CONNECTOR_DROP_TYPE,
+                    ])
+                )(
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-ignore
+                    withContextMenu(() => defaultMenu)(
+                        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                        // @ts-ignore
+
+                        withDragNode(nodeDragSourceSpec('node', true, true))(
+                            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                            // @ts-ignore
+                            withSelection()(StyleNode)
+                        )
+                    )
+                )
+            );
+        case 'node-path':
+            return CustomPathNode;
+        case 'node-polygon':
+            return CustomPolygonNode;
```

Comment:
you can get rid of `node-path` and `node-polygon` as we're not using them in this scenario